storage-proofs-porep/src/stacked/vanilla/proof.rs

```diff
@@ -100,42 +100,40 @@ impl<'a, Tree: 'static + MerkleTreeTrait, G: 'static + Hasher> StackedDrg<'a, Tr
         pub_inputs: &PublicInputs<<Tree::Hasher as Hasher>::Domain, <G as Hasher>::Domain>,
         p_aux: &PersistentAux<<Tree::Hasher as Hasher>::Domain>,
         t_aux: &TemporaryAuxCache<Tree, G>,
-        layer_challenges: &LayerChallenges,
-        layers: usize,
+        challenges: &LayerChallenges,
+        num_layers: usize,
         partition_count: usize,
     ) -> Result<Vec<Vec<Proof<Tree, G>>>> {
-        assert!(layers > 0);
+        assert!(num_layers > 0);
         // Sanity checks on restored trees.
         assert!(pub_inputs.tau.is_some());
 
-        if layer_challenges.use_synthetic {
+        if challenges.use_synthetic {
             // If there are no synthetic vanilla proofs stored on disk yet, generate them.
             if pub_inputs.seed.is_none() {
                 info!("generating synthetic vanilla proofs in a single partition");
                 assert_eq!(partition_count, 1);
 
                 let comm_r = pub_inputs.tau.as_ref().expect("tau is set").comm_r;
                 // Derive the set of challenges we are proving over.
-                let challenges = layer_challenges.derive_synthetic(
-                    graph.size(),
-                    &pub_inputs.replica_id,
-                    &comm_r,
-                );
+                let challenge_positions =
```

Comment:
At least to me, renaming `challenges` to `challenge_positions` (here and below) makes the code slightly less clear.

Reply:
Agreed, but `challenges` is what you put in as parameter, which is of type `LayerChallenges`. So I didn't want to re-assign something with the same name. I also don't really like the name, but it's the best i could come up with.

Would another name work? `challenge_offsets`? `derived`? `derived_challenges`? `derived_positions`? `derived_offsets`?

Or should we rename the input variable?

If you've any suggestions, please let me know.

Reply:
Yeah, that makes sense. Maybe here `synth_challenges` and below `partition_challenges`, or just `challenge_nodes` in both?